Jenkinsfile

```diff
@@ -342,13 +354,22 @@ class PythonParallelTaskGenerator extends DefaultParallelTaskGenerator {
                 defaultExcludes: false
               )
             }
-            steps.env.PYTHON_VERSION = "${x}"
-            steps.env.WEBFRAMEWORK = "${y}"
-            steps.codecov(repo: "${steps.env.REPO}",
-              basedir: "${steps.env.BASE_DIR}",
-              flags: "-e PYTHON_VERSION,WEBFRAMEWORK",
-              secret: "${steps.env.CODECOV_SECRET}"
-            )
+            // steps.env.PYTHON_VERSION = "${x}"
+            // steps.env.WEBFRAMEWORK = "${y}"
+            steps.dir("${steps.env.BASE_DIR}"){
+              steps.script {
+                def massaged_py_ver = steps.sh(returnStdout: true,
+                script: """
+                    docker run apm-agent-python:${x} python -c \"import platform; pv=platform.python_version_tuple(); print('pypy' + ('' if pv[0] == 2 else str(pv[0])) if platform.python_implementation() == 'PyPy' else '.'.join(map(str, platform.python_version_tuple()[:2])))\" | egrep -v 'Starting with'
```

Comment:
How would you see moving this script to a `get-version.py` script? 🤔 It would simplify this `sh` step and also would allow running the same locally, be it the case

Reply:
tbh, I think we should resolve the name clash (using `PYTHON_VERSION` environment variable twice for two different things) that makes this step necessary in the first place. Something like this should do it I think:

    diff --git a/Makefile b/Makefile
    index b30c6ca2..cfcdf881 100644
    --- a/Makefile
    +++ b/Makefile
    @@ -8,9 +8,9 @@ flake8:
            flake8
    
    test:
    -       if [[ "$$PYTHON_VERSION" =~ ^(3.6|3.7|3.8|nightly|pypy3)$$ ]] ; then \
    +       if [[ "$$PYTHON_VERSION_NUMBER" =~ ^(3.6|3.7|3.8|nightly|pypy3)$$ ]] ; then \
                    py.test -v $(PYTEST_ARGS) $(PYTEST_MARKER) $(PYTEST_JUNIT); \
    -       elif [[ "$$PYTHON_VERSION" =~ ^3\.5$$ ]] ; then \
    +       elif [[ "$$PYTHON_VERSION_NUMBER" =~ ^3\.5$$ ]] ; then \
                    py.test -v $(PYTEST_ARGS) $(PYTEST_MARKER) $(PYTEST_JUNIT) --ignore-glob='*/asyncio/*'; \
            else \
                    py.test -v $(PYTEST_ARGS) $(PYTEST_MARKER) $(PYTEST_JUNIT) --ignore-glob='*/py3_*.py' --ignore-glob='*/asyncio/*'; \
    diff --git a/tests/scripts/docker/run_tests.sh b/tests/scripts/docker/run_tests.sh
    index 70c08f40..ea5b9200 100755
    --- a/tests/scripts/docker/run_tests.sh
    +++ b/tests/scripts/docker/run_tests.sh
    @@ -54,6 +54,7 @@ PYTHON_VERSION=${1} docker-compose run \
    -e LOCAL_USER_ID=$UID \
    -e PYTHONDONTWRITEBYTECODE=1 -e WEBFRAMEWORK=$2 -e PIP_CACHE=${docker_pip_cache} \
    -e WITH_COVERAGE=true \
    +  -e PYTHON_VERSION=${1} \
    -e CASS_DRIVER_NO_EXTENSIONS=1 \
    -e PYTEST_JUNIT="--junitxml=/app/tests/python-agent-junit.xml" \
    -v ${pip_cache}:$(dirname ${docker_pip_cache}) \
    diff --git a/tests/scripts/run_tests.sh b/tests/scripts/run_tests.sh
    index 7f826b3e..0153fd69 100755
    --- a/tests/scripts/run_tests.sh
    +++ b/tests/scripts/run_tests.sh
    @@ -6,7 +6,7 @@ export PATH=${HOME}/.local/bin:${PATH}
    python -m pip install --user -U pip --cache-dir "${PIP_CACHE}"
    python -m pip install --user -r "tests/requirements/requirements-${WEBFRAMEWORK}.txt" --cache-dir "${PIP_CACHE}"
    
    -export PYTHON_VERSION=$(python -c "import platform; pv=platform.python_version_tuple(); print('pypy' + ('' if pv[0] == 2 else str(pv[0])) if platform.python_implementatio
    +export PYTHON_VERSION_NUMBER=$(python -c "import platform; pv=platform.python_version_tuple(); print('pypy' + ('' if pv[0] == 2 else str(pv[0])) if platform.python_implem